source/agora/node/Registry.d

```diff
@@ -331,32 +333,25 @@ public class NameRegistry: NameRegistryAPI
                 break;
             }
 
-            if (q.qtype == QTYPE.AXFR)
+            if (!q.qtype.among(QTYPE.A, QTYPE.CNAME, QTYPE.ALL))
             {
-                if (this.validators.matches(q.qname))
-                    this.doAXFR(this.validators, reply);
-                else if (this.flash.matches(q.qname))
-                    this.doAXFR(this.flash, reply);
-                else
-                {
-                    log.warn("Refusing AXFR for unknown zone: {}", q.qname);
-                    reply.header.RCODE = Header.RCode.Refused;
-                    break;
-                }
-            }
-            else if (q.qtype.among(QTYPE.A, QTYPE.CNAME, QTYPE.ALL))
-            {
-                reply.header.RCODE = this.getValidatorDNSRecord(q, reply);
-                if (reply.header.RCODE != Header.RCode.NoError)
-                    break;
+                log.warn("DNS: Ignoring query for unknown QTYPE: {}", q);
+                reply.header.RCODE = Header.RCode.NotImplemented;
+                break;
             }
+
+            if (this.validators.owns(q.qname))
+                this.validators.answer(q, reply);
```

Comment:
Setting `RCODE` seems a little bit scattered, `getKeyDNSRecord` returns the value even it also gets `reply` as `ref` parameter as in `answer` and `doAXFR`

Reply:
Yeah, it bothers me too, let's try to clean that up.